src/value.c

```diff
@@ -97,22 +106,91 @@ SIValue SI_Vectorf32
 	return SIVectorf32_New(dim);
 }
 
-SIValue SI_DuplicateStringVal(const char *s) {
-	return (SIValue) {
-		.stringval = rm_strdup(s), .type = T_STRING, .allocation = M_SELF
-	};
+bool USE_STRING_POOL = false; // global flag to enable/disable string pooling
+
+// check rather or not to use string pool
+inline static bool use_string_pool(void) {
```

Comment:
why to do this logic?
maybe it is better to use thread local and set it for each thread then it will be less code to check

Reply:
So the idea is to store a boolean within the `thread local storage` saying if the thread could access the `string-pool` ?